lib/order.rb

```diff
@@ -0,0 +1,53 @@
+require_relative 'menu'
+require_relative 'sms'
+
+class Order
+  attr_reader :basket
+
+  def initialize 
+    @basket = []
+    #eventually would be equal to a premade menu object (i.e Italian_restaurant) or would be able to select a premade menu
+    @menu = Menu.new   
+  end
+
+  def add_dish_to_basket(dish)
+    count = 0
+    @menu.menu.each { |food| @basket << food && count +=1 if food.name == dish }
+    raise 'There is no such dish on the menu' if count == 0
+  end
+
+  def remove_dish_from_basket(dish) 
+    matching_dishes = @basket.select { |food| food.name == dish}
+    if matching_dishes.count >= 1 
+      @basket.delete_at(@basket.index(matching_dishes[0])) 
+    else
+      raise "This dish is not currently in your basket"
+    end
+  end
+
+  def calculate_order_price
+    @cost = 0
+    @basket.each { |dish| @cost += dish.price }
+    if basket.empty?
+      puts "Your basket is empty"
```

Comment:
Ideally this method shouldn't print out messages with `puts` - because we'd expect this method to only calculate and return the total price (from its name `calculate_order_price`). If we wanted to print out messages, this would probably belong to a different class, which would be responsible for presenting the data of the program to the user, something like `OrderPrinter`